### PR TITLE
Add support for -p/--parents in mkdir

### PR DESCRIPTION
**Pull request by innerout**

### CODE REVIEW

File: applications/coreutils/mkdir.cpp [hunk 1]
```diff
@@ -1,17 +1,58 @@
...
 int main(int argc, char **argv)
 {
     if (argc == 2)
     {
         return mkdir(argv[1]);
     }
 
+    if (argc == 3)
+    {
+        if (!strcmp(argv[1], "-p") || !strcmp(argv[1], "--parents"))
```
Comment (Keyboard-Slayer) [inline applications/coreutils/mkdir.cpp:52]: Hey nice work :+1:

But you should use the header `libsystem/cmdline/CMDLine.h` for parsing the arguments 
If you want some examples you can look at [uname](https://github.com/skiftOS/skift/blob/master/applications/coreutils/uname.cpp) and [ls](https://github.com/skiftOS/skift/blob/master/applications/coreutils/ls.cpp)

**Review by Keyboard-Slayer — CHANGES_REQUESTED** [review]

File: applications/coreutils/mkdir.cpp (hunk 1 above)
Comment (innerout) [reply applications/coreutils/mkdir.cpp]: Resolved it, should I change anything else?

File: applications/coreutils/mkdir.cpp [hunk 2]
```diff
@@ -41,17 +60,17 @@ void mkdir_parent_dirs(const char *path)
...
 int main(int argc, char **argv)
-{
+{    
     if (argc == 2)
     {
         return mkdir(argv[1]);
     }
 
-    if (argc == 3)
+    argc = cmdline_parse(&cmdline, argc, argv);
```
Comment (sleepy-monax) [inline applications/coreutils/mkdir.cpp:69]: cmdline_parse should be called before everything because he will consume some argv values?

**Review by sleepy-monax — CHANGES_REQUESTED** [review]

File: applications/coreutils/mkdir.cpp (hunk 2 above)
Comment (innerout) [reply applications/coreutils/mkdir.cpp:69]: How should I check for the case `mkdir folderpath` using cmdline_parse?

Comment (sleepy-monax) [reply applications/coreutils/mkdir.cpp:69]: cmdline_parse will remove all option like -p and won't touch the rest ;)

Comment (sleepy-monax) [reply applications/coreutils/mkdir.cpp:69]: applications/coreutils/ls.cpp is a good exemple ;)

File: applications/coreutils/mkdir.cpp [hunk 3]
```diff
@@ -1,17 +1,92 @@
...
+#include <stdio.h>
+#include <unistd.h>
+
+static char * parent = NULL;
+
+static CommandLineOption options[] = {
+    COMMANDLINE_OPT_HELP,
+    COMMANDLINE_OPT_STRING("parents", 'p', parent, "Make parent directories as needed.", COMMANDLINE_NO_CALLBACK),
+    COMMANDLINE_OPT_END};
+   //COMMANDLINE_OPT_STRING("", '', dir, "Directory to create.", COMMANDLINE_NO_CALLBACK),
```
Comment (sleepy-monax) [inline applications/coreutils/mkdir.cpp:14]: Don't push commented out code

**Review by sleepy-monax — CHANGES_REQUESTED** [review]

**Comment by sleepy-monax** [comment]
Looks goods I'm merging that :)

**Pull request merged**